internal/state/registry.go

```diff
@@ -2,68 +2,94 @@ package state
 
 import (
 	"context"
-	"github.com/pkg/errors"
-
-	"github.com/kyma-project/serverless-manager/internal/registry"
+	"fmt"
 
 	"github.com/kyma-project/serverless-manager/api/v1alpha1"
 	"github.com/kyma-project/serverless-manager/internal/chart"
+	"github.com/kyma-project/serverless-manager/internal/registry"
+	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func sFnRegistryConfiguration(ctx context.Context, r *reconciler, s *systemState) (stateFn, *ctrl.Result, error) {
-	switch {
-	case *s.instance.Spec.DockerRegistry.EnableInternal:
-		err := setInternalRegistry(ctx, r, s)
-		if err != nil {
-			s.setState(v1alpha1.StateError)
-			s.instance.UpdateConditionFalse(
-				v1alpha1.ConditionTypeConfigured,
-				v1alpha1.ConditionReasonConfigurationErr,
-				err,
-			)
-			return nextState(sFnUpdateStatusWithError(err))
-		}
-	case s.instance.Spec.DockerRegistry.SecretName != nil:
-		err := setExternalRegistry(ctx, r, s)
-		if err != nil {
-			s.setState(v1alpha1.StateError)
-			s.instance.UpdateConditionFalse(
-				v1alpha1.ConditionTypeConfigured,
-				v1alpha1.ConditionReasonConfigurationErr,
-				err,
-			)
-			return nextState(sFnUpdateStatusWithError(err))
-		}
-	default:
-		setK3dRegistry(s)
+	// setup status.dockerRegistry and set possible warnings
+	err := configureRegistry(ctx, r, s)
+	if err != nil {
+		s.setState(v1alpha1.StateError)
+		s.instance.UpdateConditionFalse(
+			v1alpha1.ConditionTypeConfigured,
+			v1alpha1.ConditionReasonConfigurationErr,
+			err,
+		)
+		return nextState(sFnUpdateStatusWithError(err))
 	}
-	// overwrite registryStatus when there is serverless (not serverless-manager) cluster-wide external registry secret
-	setExternalRegistrySecretNameInDockerRegistryStatus(ctx, r, s)
 
+	// update status if needed
 	if s.snapshot.DockerRegistry != s.instance.Status.DockerRegistry {
 		return nextState(sFnUpdateStatusAndRequeue)
 	}
 
 	return nextState(sFnOptionalDependencies)
 }
 
-func setExternalRegistrySecretNameInDockerRegistryStatus(ctx context.Context, r *reconciler, s *systemState) {
-	// doc: https://kyma-project.io/docs/kyma/latest/05-technical-reference/svls-03-switching-registries#cluster-wide-external-registry
+func configureRegistry(ctx context.Context, r *reconciler, s *systemState) error {
 	secret, err := registry.GetServerlessExternalRegistrySecret(ctx, r.client, s.instance.GetNamespace())
-	// ignore errors because it only set status
-	if err != nil || secret == nil {
-		return
+	if err != nil {
+		return err
 	}
-	if address, ok := secret.Data["serverAddress"]; ok {
-		s.instance.Status.DockerRegistry = string(address)
+
+	switch {
+	case secret != nil:
```

Comment:
I would name it as `extRegSecert` for clarity, as it's not ordinary secret

Reply:
Done